app/models/bottleneck_event.rb

```diff
@@ -47,6 +47,25 @@ def self.calculate_future_event(obj, options)
     send(method, obj, options)
   end
 
+  def self.build_event_message(context)
+    value =
+      case context[:event][:event_type]
+      when "CpuUsage"
+        MiqReport.new.format_mhz_to_human_size(context[:limit_attr_value], :precision => 1)
+      when "MemoryUsage"
+        MiqReport.new.format_bytes_to_human_size(context[:limit_attr_value].megabytes, :precision => 1)
+      when "DiskUsage"
+        MiqReport.new.format_bytes_to_human_size(context[:limit_attr_value], :precision => 1)
+      end
+
+    "%{trend_attr} is projected to reach %{limit_value} (%{limit_pct}%% of %{limit_attr})" % {
+      :trend_attr  => Dictionary.gettext(context[:calculation][:trend_attr].to_s, :type => "column"),
+      :limit_value => value,
+      :limit_pct   => context[:calculation][:limit_pct],
+      :limit_attr  => Dictionary.gettext(context[:calculation][:limit_attr].to_s, :type => "column")
+    }
+  end
+
   def self.event_definitions(event_type)
     @event_definitions ||= {}
     @event_definitions[event_type] ||= MiqEventDefinition.where(:event_type => event_type).to_a
```

Comment:
Total aside - this class-caching feels problematic to me.

Reply:
yeah, seems like a premature optimization to me.  I'm not sure why it needs to be cached.